wordpress-mobile/WordPress-Android
# My Site Dashboard: Tabs - Implement default tab experiment logic + Show QS Card on default tab

Closes #16180

This PR 

1. Sets initial `My Site` tab based on the default tab experiment variant as follows:
If `AppPrefs.MY_SITE_DEFAULT_TAB_EXPERIMENT_VARIANT` == "dashboard" => default to DASHBOARD
Otherwise, `SITE MENU` is set as the default tab.

2. Displays `Quick Start Card` on the initial default tab. For this, I removed the hard-coded `QuickStartOrigin` in the `QuickStartTasksRepository` and used the default tab value in the `MySiteViewModel` for displaying the card in 4e60687897babe46667c8672561fdcb0c46c50f5.

To test:

`Test.1 - Default Tab Selection`
- Since we cannot control which value gets returned from the experiment for the default tab, I relied on the unit tests for testing the functionality. Make sure that the unit tests added in b6f906c3efecc65fb4e3d839c98cd6041d8c0013 cover important scenarios. 
Another option is to manually hard-code the `MySiteTabExperimentVariant` value in `AppPrefs.getMySiteDefaultTabExperimentVariant` and check that appropriate tab is shown as the default tab. 

`Test.2 - Quick Start Card on Default Tab`
- I updated existing tests that were dependent on `QuickStartOrigin` to now use the `MySiteDefaultTabExperiment` value. Make sure that the unit tests in d6c3b57c80c0c9ebacd2991fa19112e2df9bbec7 are updated correctly. 

`Test.3 - Quick Start Flow`
We only tested `Quick Start - Site Menu Step` when the quick start card was shown on the `Dashboard`/ `Home` tab. Now it is possible that the card is shown on `Site Menu` tab, so we might want to retest QS flow from QS card on this tab.

- Manually hard-code the `MySiteTabExperimentVariant` value to SITE_MENU variant in `AppPrefs.getMySiteDefaultTabExperimentVariant` .
- Reinstall the app.
- Go to My Site -> Me -> App Settings -> Debug settings.
- Turn on the `MySiteDashboardTabsFeatureConfig` flag under the Features in development section.
- Logout/ login and select "Show me around" when the quick start prompt is shown.
- Notice that `Quick Start Card` is shown in the `Site Menu` tab. 
- Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt). 

## Regression Notes
1. Potential unintended areas of impact
The initial `My Site` tab is not set correctly. 
`Quick Start Card` is not shown on the initial tab and `Quick Start` flows don't work as expected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I relied on automated tests as specified in the `To test` section.

4. What automated tests I added (or what prevented me from doing so)
Added/ updated unit tests to test default tab based on default tab experiment variant and that quick start is shown on the default tab.

PR submission checklist:
